flink-kubernetes/pom.xml

```diff
@@ -0,0 +1,120 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+		 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.apache.flink</groupId>
+		<artifactId>flink-parent</artifactId>
+		<version>1.9-SNAPSHOT</version>
+		<relativePath>..</relativePath>
+	</parent>
+
+	<artifactId>flink-kubernetes_${scala.binary.version}</artifactId>
+	<name>flink-kubernetes</name>
+	<packaging>jar</packaging>
+
+	<properties>
+		<kubernetes.version>1.0.1</kubernetes.version>
+	</properties>
+
+	<dependencies>
+
+		<!-- set all Flink dependencies to provided, so they and their transitive  -->
+		<!-- dependencies do not get promoted to direct dependencies during shading -->
+
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-clients_${scala.binary.version}</artifactId>
+			<version>${project.version}</version>
+			<scope>provided</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-runtime_${scala.binary.version}</artifactId>
+			<version>${project.version}</version>
+			<scope>provided</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-core</artifactId>
+			<version>2.7.5</version>
+		</dependency>
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-databind</artifactId>
+			<version>2.7.5</version>
+		</dependency>
+
+		<dependency>
+			<groupId>io.fabric8</groupId>
+			<artifactId>kubernetes-client</artifactId>
```

Comment:
We might wanna relocate the transitive dependencies of this dependency so that user code won't be affected by them (dependency conflicts) (e.g. relocating the jackson dependencies).

Reply:
Previously I met the build error in travis like this(which not occur in my machine):
`Dependency convergence error for com.fasterxml.jackson.core:jackson-databind:2.7.5 paths to dependency are:
+-org.apache.flink:flink-kubernetes_2.11:1.9-SNAPSHOT
  +-io.fabric8:kubernetes-client:4.0.0
    +-io.fabric8:kubernetes-model:3.0.0
      +-com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.7.5
        +-com.fasterxml.jackson.core:jackson-databind:2.7.5
and
+-org.apache.flink:flink-kubernetes_2.11:1.9-SNAPSHOT
  +-io.fabric8:kubernetes-client:4.0.0
    +-com.fasterxml.jackson.core:jackson-databind:2.7.7
and
+-org.apache.flink:flink-kubernetes_2.11:1.9-SNAPSHOT
  +-io.fabric8:kubernetes-client:4.0.0
    +-io.fabric8:zjsonpatch:0.3.0
      +-com.fasterxml.jackson.core:jackson-databind:2.3.2`

Seems kubernetes-client has several jackson version which fail the build. So I exclude jackson dependencies in kubenetes-client and include dependency at the mean while. In this way, the problem is resolved.

I don‘t quite understand “relocate the transitive dependencies”, does it mean we specify versions of transitive dependencies in DependencyManagement？I updated this PR in this way, and the dependency error problem could also be resolved.

Reply:
If you wanna fix a certain transitive dependency to a certain version, then it is enough to explicitly list this dependency as a direct dependency. Maven's dependency conflict resolution mechanism will use the dependency which is closest to the project root.

With relocating transitive I mean to give them a new namespace so that other code won't see it. The process is described here https://maven.apache.org/plugins/maven-shade-plugin/examples/class-relocation.html and we use it in various modules in Flink: https://github.com/apache/flink/blob/master/flink-shaded-hadoop/pom.xml#L146.

Reply:
Thanks for your pointer, we need add relocate jackson related dependencied with adding  something like <relocation> in this pom, right?